tables/cost_and_usage_report/cost_and_usage_report_table.go

```diff
@@ -0,0 +1,111 @@
+package cost_and_usage_report
+
+import (
+	"strings"
+	"time"
+
+	"github.com/rs/xid"
+	typehelpers "github.com/turbot/go-kit/types"
+	"github.com/turbot/pipe-fittings/v2/utils"
+	"github.com/turbot/tailpipe-plugin-aws/sources/s3_bucket"
+
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source_config"
+	"github.com/turbot/tailpipe-plugin-sdk/constants"
+	"github.com/turbot/tailpipe-plugin-sdk/row_source"
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+	"github.com/turbot/tailpipe-plugin-sdk/table"
+)
+
+const CostUsageReportTableIdentifier = "aws_cost_and_usage_report"
+
+// CostUsageReportTable - table for CostUsageReports
+type CostUsageReportTable struct{}
+
+// Identifier implements table.Table
+func (t *CostUsageReportTable) Identifier() string {
+	return CostUsageReportTableIdentifier
+}
+
+func (t *CostUsageReportTable) GetSourceMetadata() []*table.SourceMetadata[*CostUsageReport] {
+	defaultS3ArtifactConfig := &artifact_source_config.ArtifactSourceConfigImpl{
+		FileLayout: utils.ToStringPointer("%{DATA:export_name}/(?:data/%{DATA:partition}/)?(?:%{INT:from_date}-%{INT:to_date}/)?(?:%{DATA:assembly_id}/)?(?:%{DATA:timestamp}-%{DATA:execution_id}/)?%{DATA:file_name}.csv.(?:zip|gz)"),
```

Comment:
Can you please add a comment here showing the different file patterns we want to capture here (for legacy + CUR 2.0)?